validmind/vm_models/test/metric_v2.py

```diff
@@ -0,0 +1,120 @@
+# Copyright © 2023-2024 ValidMind Inc. All rights reserved.
+# See the LICENSE file in the root of this repository for details.
+# SPDX-License-Identifier: AGPL-3.0 AND ValidMind Commercial
+
+"""
+Class for storing ValidMind metric objects and associated
+data for display and reporting purposes
+"""
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import ClassVar, Optional, Union
+
+import pandas as pd
+
+from validmind.vm_models import MetricResult
+
+from ...utils import clean_docstring
+from .result_wrapper import MetricResultWrapper
+from .test import Test
+
+
+@dataclass
+class UnitMetric(Test):
+    """
+    Metric objects track the schema supported by the ValidMind API
+    """
+
+    # Class Variables
+    test_type: ClassVar[str] = "UnitMetric"
+
+    type: ClassVar[str] = ""  # type of metric: "training", "evaluation", etc.
+    scope: ClassVar[str] = ""  # scope of metric: "training_dataset"
```

Comment:
This was legacy from the metric class, which we don't use anymore. We use `metadata`. Let's remove this (and its usage in the unit metrics on this PR).

Reply:
Done.